mdz/Makefile

```diff
@@ -137,31 +134,23 @@ swag-install:
 	go install github.com/swaggo/swag/cmd/swag@v1.8.7
 
 build-local:
-	@for target in $(TARGETS); do                                                      \
-	  CGO_ENABLED=$(CGO_ENABLED) GOOS=$(GOOS) GOARCH=$(GOARCH) go build -trimpath -v -o $(OUTPUT_DIR)/$${target}     \
-	    -ldflags "-s -w -X $(ROOT)/pkg/version.version=$(VERSION) -X $(ROOT)/pkg/version.buildDate=$(BUILD_DATE) -X $(ROOT)/pkg/version.gitCommit=$(GIT_COMMIT) -X $(ROOT)/pkg/version.gitTreeState=$(GIT_TREE_STATE)"                     \
-	    $(CMD_DIR)/$${target};                                                         \
-	done
+	CGO_ENABLED=$(CGO_ENABLED) GOOS=$(GOOS) GOARCH=$(GOARCH) \
+	VERSION=$(VERSION) BUILD_DATE=$(BUILD_DATE) \
+	GIT_COMMIT=$(GIT_COMMIT) GIT_TREE_STATE=$(GIT_TREE_STATE) GIT_TAG=$(GIT_TAG) \
+	./hack/build.sh build
```

Comment:
Does this mean the `make build-local` will use the remote agent version?

Reply:
There is a hack in the build.sh, it should use local version if I understand well.

/cc @zwpaper

Reply:
only `debug-local` will use local agent, both `make build-local` and `make build` will use the remote.

I previously thought we should use the `make build-local` to use the local agent, but I found a `debug-local` when implementation, and it seems to be a better choice

Reply:
I thinks debug-local is for vscode breakpoints.

Reply:
Or maybe we make it
- build: locally built with remote deps
- Build-local: locally built with local deps
- build-release: build with tag info
- debug: debug build with remote deps
- Debug-local: debug build with local deps

It seems a bit complicated...

Reply:
I have a new idea, we add an env var for local deps, like

`DEPS=local make build` will use local deps, otherwise will use remote deps

Reply:
IMO `make` commands are mainly used for development. The release procedure is done by the CI. Thus the commands should prioritise the development experience. It's very common to build with local `openmodelz/agent` during development since you will change the code frequently.

1. debug should always build with local
2. not sure if we can build in `/tmp` env and output to repo's `/bin`, I want to avoid changing the repo's `go.mod` otherwise I need the revert it before the git commit

Reply:
No, I don't think so, Make is not only for debug.

I agree with you on it would be convenient to debug with local dependencies, but not like Rust, local dependencies is never a good option in Go.

What we're discussing here is to make local debugging easier.

the env var like `DEPS=local make build` seems to be a good choice IMHO.

Reply:
The context here is that you won't need to build the release by yourself locally. It's already done in the CI.

It's not a local dependencies, it's dev with local changes. We won't release the package with local dependencies.

This PR makes it easy to use `go install` (recommend using pip since some go versions will break the code) but also harms local development. Building `mdz` with local agent code is necessary. Otherwise, you won't be able to add new features.

Reply:
let's discuss it here https://github.com/tensorchord/openmodelz/issues/147